cloudsync/cs.py

```diff
@@ -273,11 +273,10 @@ def stop(self, forever=True, wait=True):
         self.emgrs[0].stop(forever=forever, wait=wait)
         self.emgrs[1].stop(forever=forever, wait=wait)
         self.nmgr.stop(forever=forever, wait=wait)
-        if self.sthread:
+        if self.sthread and wait:
```

Comment:
Please refactor this so that `join()`ing the threads is required, or not joining prints a warning. Either

- get rid of wait here, create a utility method that takes a wait flag and multiple cs's to stop them all at once, or
- track if join was ever called, and if not, print a warning before the program quits to identify the issue
- use some structure that allows for shutting this down in the background, that warns if there isn't followup, kind of like a `future`
- some other solution that you like

It's important to ensure that the threads actually complete, or hard-to-diagnose bugs will be introduced.

Reply:
done.